python/revng/internal/cli/_commands/generate_migrations/__init__.py

```diff
@@ -0,0 +1,239 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import subprocess
+from pathlib import Path
+from typing import Dict, List, Literal, Tuple
+
+import jinja2
+import jsonschema
+import yaml
+
+from revng.internal.cli.commands_registry import Command, CommandsRegistry
+from revng.internal.cli.support import log_error
+
+
+class GenerateMigrationException(Exception):
+    pass
+
+
+class GenerateMigrationCommand(Command):
+    def __init__(self):
+        super().__init__(
+            ("model", "generate-migrations"),
+            "Generate migrations for a schema based on its git history",
+        )
+
+    def register_arguments(self, parser):
+        parser.add_argument("schema", help="The path to the schema.")
+        parser.add_argument(
+            "--output", help="Output the migration template here, default to stdout"
+        )
+
+    def _get_metaschema(self):
+        metaschema_path = Path(__file__).parent / "metaschema.yml"
+
+        with open(metaschema_path) as file:
+            metaschema = yaml.safe_load(file)
+
+        return metaschema
+
+    def _get_raw_input(self, schema_path: str) -> str:
+        try:
+            with open(schema_path) as file:
+                raw_input = file.read()
+        except OSError as error:
+            raise GenerateMigrationException(f"Error while reading input from file: {error}")
+        return raw_input
+
+    def _get_raw_input_develop_branch(self, schema_path: str) -> str:
+        try:
+            git_process = subprocess.run(
+                f"git --no-pager show develop:{schema_path}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+        except subprocess.CalledProcessError as error:
+            raise GenerateMigrationException(
+                f"Error while running git: {error}\n{error.stderr.decode().strip()}"
+            )
+
+        raw_schema = git_process.stdout.decode()
+
+        return raw_schema
+
+    def _restructure(self, schema: Dict) -> Dict:
+        """Restructures the schema into a nested dictionaries where the `Name` field of types and
+        fields/members are keys.
+        """
+        result: Dict = {}
```

Comment:
Is dict the best we can say here?

Reply:
I suppose `Dict[str, Any]` is as good as we can go without invoking the `metaschema.yml`.

Note: I just realized `typing.Dict` is deprecated since Python 3.9, `builtins.dict` is the intended one now. Should I use that instead?

Reply:
> I suppose Dict[str, Any] is as good as we can go without invoking the metaschema.yml.

OK.

> Note: I just realized typing.Dict is deprecated since Python 3.9, builtins.dict is the intended one now. Should I use that instead?

See what we're doing elsewhere and be coherent.